applications/FluidDynamicsApplication/custom_elements/surface_smoothing_element.cpp

```diff
@@ -0,0 +1,670 @@
+ * note: time integration parameters must be set in the rCurrentProcessInfo before calling these methods
+ * CalculateSecondDerivativesContributions,
+ * CalculateSecondDerivativesLHS, CalculateSecondDerivativesRHS methods are : OPTIONAL
+ */
+
+
+/**
+ * this is called during the assembling process in order
+ * to calculate the second derivative contributions for the LHS and RHS
+ * @param rLeftHandSideMatrix: the elemental left hand side matrix
+ * @param rRightHandSideVector: the elemental right hand side
+ * @param rCurrentProcessInfo: the current process info instance
+ */
+void SurfaceSmoothingElement::CalculateSecondDerivativesContributions(
+    MatrixType& rLeftHandSideMatrix,
+    VectorType& rRightHandSideVector,
+    ProcessInfo& rCurrentProcessInfo)
+{
+    if (rLeftHandSideMatrix.size1() != 0)
+        rLeftHandSideMatrix.resize(0, 0, false);
+    if (rRightHandSideVector.size() != 0)
+        rRightHandSideVector.resize(0, false);
+}
+
+/**
+ * this is called during the assembling process in order
+ * to calculate the elemental left hand side matrix for the second derivatives constributions
+ * @param rLeftHandSideMatrix: the elemental left hand side matrix
+ * @param rCurrentProcessInfo: the current process info instance
+ */
+void SurfaceSmoothingElement::CalculateSecondDerivativesLHS(
+    MatrixType& rLeftHandSideMatrix,
+    ProcessInfo& rCurrentProcessInfo)
+{
+    if (rLeftHandSideMatrix.size1() != 0)
+        rLeftHandSideMatrix.resize(0, 0, false);
+}
+
+/**
+ * this is called during the assembling process in order
+ * to calculate the elemental right hand side vector for the second derivatives constributions
+ * @param rRightHandSideVector: the elemental right hand side vector
+ * @param rCurrentProcessInfo: the current process info instance
+ */
+void SurfaceSmoothingElement::CalculateSecondDerivativesRHS(
+    VectorType& rRightHandSideVector,
+    ProcessInfo& rCurrentProcessInfo)
+{
+    if (rRightHandSideVector.size() != 0)
+        rRightHandSideVector.resize(0, false);
+}
+
+/**
+ * this is called during the assembling process in order
+ * to calculate the elemental mass matrix
+ * @param rMassMatrix: the elemental mass matrix
+ * @param rCurrentProcessInfo: the current process info instance
+ */
+void SurfaceSmoothingElement::CalculateMassMatrix(MatrixType& rMassMatrix, ProcessInfo& rCurrentProcessInfo)
+{
+    if (rMassMatrix.size1() != 0)
+        rMassMatrix.resize(0, 0, false);
+}
+
+/**
+ * this is called during the assembling process in order
+ * to calculate the elemental damping matrix
+ * @param rDampingMatrix: the elemental damping matrix
+ * @param rCurrentProcessInfo: the current process info instance
+ */
+void SurfaceSmoothingElement::CalculateDampingMatrix(MatrixType& rDampingMatrix, ProcessInfo& rCurrentProcessInfo)
+{
+    if (rDampingMatrix.size1() != 0)
+        rDampingMatrix.resize(0, 0, false);
+}
```

Comment:
Why these are required to be override?

Reply:
> Sorry if I'm adding noise, but I wanted to ask the following:
> 
> * How does this process differ to the VariationalDistanceProcess? In a practical sense, is it faster? more accurate?

Hi @marcnunezc 
The main aim of the variational distance process is to keep the distance field practically usable (preventing DISTANCE variable deviating too much from real/expected distance).
This process filters out the noises in the DISTANCE field (in fact noises are diffused).

Reply:
> Why these are required to be override?

I agree those are not needed, I am quite sure the baseclass does the same. Can you check @mrhashemi ?

Reply:
I removed them here.